src/main/java/uk/gov/pay/webhooks/deliveryqueue/managed/WebhookMessageSendingQueueProcessor.java

```diff
@@ -135,7 +85,7 @@ private void pollQueue() {
         Transaction transaction = session.beginTransaction();
         try {
             Optional<WebhookDeliveryQueueEntity> maybeQueueItem = webhookDeliveryQueueDao.nextToSend(Date.from(instantSource.instant()));
-            maybeQueueItem.ifPresent(this::attemptSend);
+            maybeQueueItem.ifPresent(item -> new SendAttempter(webhookDeliveryQueueDao, instantSource, new WebhookMessageSender(httpClient, objectMapper, webhookMessageSignatureGenerator)).attemptSend(item));
```

Comment:
If I understand correctly, this is instantiating two fairly involved objects, per webhook message delivery attempt. I don't think this will be an issue for low volumes but given peak traffic I could see the time/ complexity adding up.

As this class is importing a lot of things and then just passing them down into instantiations it feels like these should use the same singleton pattern and have those imports injected further down rather than being passed down through instantiation.

Practically I think the `SendAttempter` can be injected as a singleton into this object, it can then be responsible for importing the `ObjectMapper`, `WebhookMessageSignatureGenerator` and `HTTPClient` (none of which are actually used in this object). 

I think the same kind of analysis can be done on the `httpClient` which is then initialised by the send attempter. I assume it's fine to be using the same http client for all of these requests but that's something that would need validating.

Reply:
@sfount Good points, hopefully have addressed in 738b3a0